bliss/surveys/dc2.py

```diff
@@ -293,6 +183,161 @@ def test_dataloader(self):
+            "wcs": result_dict["other_info"]["wcs"],
+            "psf_params": result_dict["inputs"]["psf_params"],
+        }
+
+
+def squeeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "1 h w nh nw -> h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "1 h w -> h w")
+    return tile_dict_copy
+
+
+def unsqueeze_tile_dict(tile_dict):
+    tile_dict_copy = copy.copy(tile_dict)
+    for k, v in tile_dict_copy.items():
+        if k != "n_sources":
+            tile_dict_copy[k] = rearrange(v, "h w nh nw -> 1 h w nh nw")
+    tile_dict_copy["n_sources"] = rearrange(tile_dict_copy["n_sources"], "h w -> 1 h w")
+    return tile_dict_copy
+
+
+def load_image_and_catalog(
+    image_idx, image_files, bg_files, image_lim, bands, n_bands, cat_path, min_flux_threshold
+):
+    image_list, bg_list, wcs_header_str = read_frame_for_band(
+        image_files, bg_files, image_idx, n_bands, image_lim
+    )
+    wcs = from_wcs_header_str_to_wcs(wcs_header_str)
+    image = np.stack(image_list)
+    bg = np.stack(bg_list)
+
+    plocs_lim = image[0].shape
+    height = plocs_lim[0]
+    width = plocs_lim[1]
+    full_cat, psf_params, match_id = Dc2FullCatalog.from_file(
+        cat_path, wcs, height, width, bands, min_flux_threshold
+    )
+    tile_dict = full_cat.to_tile_catalog(4, 5).get_brightest_sources_per_tile().to_dict()
+    tile_dict = squeeze_tile_dict(tile_dict)
+    tile_dict["star_fluxes"] = tile_dict["star_fluxes"].clamp(min=1e-18)
+    tile_dict["galaxy_fluxes"] = tile_dict["galaxy_fluxes"].clamp(min=1e-18)
+    tile_dict["galaxy_params"][..., 3] = tile_dict["galaxy_params"][..., 3].clamp(min=1e-18)
+    tile_dict["galaxy_params"][..., 5] = tile_dict["galaxy_params"][..., 5].clamp(min=1e-18)
+
+    return {
+        "tile_dict": tile_dict,
+        "inputs": {
+            "image": image,
+            "bg": bg,
+            "psf_params": psf_params,
+        },
+        "other_info": {
+            "full_cat": full_cat,
+            "wcs": wcs,
+            "wcs_header_str": wcs_header_str,
+            "match_id": match_id,
+        },
+    }
+
+
+def generate_split_file(image_index, self_copy):
+    split_count = 0
+    result_dict = load_image_and_catalog(
+        image_index,
+        self_copy["image_files"],
+        self_copy["bg_files"],
+        self_copy["image_lim"],
+        self_copy["bands"],
+        self_copy["n_bands"],
+        self_copy["cat_path"],
+        self_copy["min_flux_threshold"],
+    )
+
+    image = result_dict["inputs"]["image"]
+    bg = result_dict["inputs"]["bg"]
+    tile_dict = result_dict["tile_dict"]
+    wcs_header_str = result_dict["other_info"]["wcs_header_str"]
+    psf_params = result_dict["inputs"]["psf_params"]
+
+    # split image
+    split_lim = self_copy["image_lim"][0] // self_copy["n_split"]
```

Comment:
Could you add a small test of the new code to `test_dc2.py`? It wouldn't need to assert anything: it could just run the code to check for syntax errors. (A "smoke test".) To make the test work, you may need to add a few representative DC2 files to the `data/tests/dc2` directory.

You can see all the code that currently isn't exercised by pytest here:
https://app.codecov.io/gh/prob-ml/bliss/blob/yd%2Fdc2_new_plots/bliss%2Fsurveys%2Fdc2.py

Reply:
if we can run the `prepare_data` function of DC2, we can cover most of the code. however, in previous PR, i found that (probably) the `multiprocessing` called by `generate_split_file` can't run in our test server. it will lead to dead lock. I'll try to bypass this limit

Reply:
Maybe we could set the number of processes to launch in the hydra config file, and we could set it to 1 in the testing config?

Reply:
i'll try it